migrations/versions/c76670622e1e_add_impact_score.py

```diff
@@ -0,0 +1,123 @@
+"""add_impact_score.
+
+Revision ID: c76670622e1e
+Revises: e3cb5f234bbf
+Create Date: 2022-07-13 15:11:52.699325
+
+"""
+import json
+import warnings
+
+import pandas as pd
+import sqlalchemy as sa
+from alembic import op
+
+warnings.filterwarnings("ignore")
+
+# revision identifiers, used by Alembic.
+revision = "c76670622e1e"
+down_revision = "e3cb5f234bbf"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    """Upgrade anomaly_data_output to add impact column."""
+    # ### commands auto generated by Alembic - please adjust! ###
+
+    conn = op.get_bind()
+
+    df = pd.read_sql("SELECT * from anomaly_data_output;", conn)
+    df["data_datetime"] = pd.to_datetime(df["data_datetime"])
+    df["series_type"] = df["series_type"].apply(
+        lambda series: json.dumps(series) if series else None
+    )
+    df_with_metrics = pd.DataFrame()
+
+    kpi_ids = df["kpi_id"].unique()
+    for kpi_id in kpi_ids:
+        # split data based on kpi id
+        df_kpi = df.loc[df["kpi_id"] == kpi_id].reset_index(drop=True)
+        # fetch anomaly period from kpi table
+        anom_period_query_result = conn.execute(
+            f"SELECT anomaly_params ->> 'anomaly_period' FROM kpi WHERE id={kpi_id}"
+        )
+        for row in anom_period_query_result:
+            anom_period = int(row[0])
+
+        df_kpi_with_metric = _calculate_metric(df_kpi, anom_period)
+        df_with_metrics = df_with_metrics.append(df_kpi_with_metric, ignore_index=True)
+
+    conn.execute("TRUNCATE TABLE anomaly_data_output;")
+    op.add_column(
+        "anomaly_data_output",
+        sa.Column("impact", sa.Float(), autoincrement=False, nullable=True),
+    )
+    df_with_metrics.to_sql("anomaly_data_output", conn, if_exists="append", index=False)
```

Comment:
Because the table already exists.
`if_exists` is used to tell how to behave if the table already exists. `"append"` will insert new values to the existing table. 
If I remove the field I get a ValueError: `ValueError: Table 'anomaly_data_output' already exists.`

Reply:
Ah ok. Got it